e2e-tests/IIoTPlatform-E2E-Tests/Orchestrated/T_TwinBrowseOrchestratedTestTheory.cs

```diff
@@ -0,0 +1,175 @@
+            _context = context ?? throw new ArgumentNullException(nameof(context));
+            _context.OutputHelper = output ?? throw new ArgumentNullException(nameof(output));
+        }
+
+        [Fact, PriorityOrder(100)]
+        public void A1_SwitchToOrchestratedMode() {
+            TestHelper.SwitchToOrchestratedModeAsync(_context).GetAwaiter().GetResult();
+        }
+
+        [Fact, PriorityOrder(101)]
+        public void A2_RegisterOPCServer_And_ActivateEndpoint() {
+            var cts = new CancellationTokenSource(TestConstants.MaxTestTimeoutMilliseconds);
+
+            // We will wait for microservices of IIoT platform to be healthy and modules to be deployed.
+            TestHelper.WaitForServicesAsync(_context, cts.Token).GetAwaiter().GetResult();
+            _context.RegistryHelper.WaitForIIoTModulesConnectedAsync(_context.DeviceConfig.DeviceId, cts.Token).GetAwaiter().GetResult();
+
+            var simulatedOpcServer = TestHelper.GetSimulatedPublishedNodesConfigurationAsync(_context, cts.Token).GetAwaiter().GetResult();
+            var testPlc = simulatedOpcServer.Values.First();
+
+            TestHelper.Registry_RegisterServerAsync(_context, testPlc.EndpointUrl, cts.Token).GetAwaiter().GetResult();
+
+            _context.ConsumedOpcUaNodes[testPlc.EndpointUrl] = _context.GetEntryModelWithoutNodes(testPlc);
+            dynamic json = TestHelper.WaitForDiscoveryToBeCompletedAsync(_context, cts.Token, new List<string> { testPlc.EndpointUrl }).GetAwaiter().GetResult();
+
+            var numberOfItems = (int)json.items.Count;
+            bool found = false;
+
+            for (int indexOfTestPlc = 0; indexOfTestPlc < numberOfItems; indexOfTestPlc++) {
+
+                var endpoint = ((string)json.items[indexOfTestPlc].discoveryUrls[0]).TrimEnd('/');
+                if (TestHelper.IsUrlStringsEqual(endpoint, testPlc.EndpointUrl)) {
+                    found = true;
+                    break;
+                }
+            }
+
+            Assert.True(found, "OPC Application not activated");
+
+            if (string.IsNullOrWhiteSpace(_context.OpcUaEndpointId)) {
+                WaitForEndpointDiscoveryAndStoreInContext();
+                Assert.False(string.IsNullOrWhiteSpace(_context.OpcUaEndpointId), "The endpoint was not set");
+            }
+
+            TestHelper.Registry_ActivateEndpointAsync(_context, cts.Token).GetAwaiter().GetResult();
+        }
+
+        [Fact, PriorityOrder(102)]
+        public void D1_Discover_OPC_UA_Endpoints() {
+            var cts = new CancellationTokenSource(TestConstants.MaxTestTimeoutMilliseconds);
+
+            var endpoints = TestHelper.Registry_GetEndpointsAsync(_context, cts.Token).GetAwaiter().GetResult();
+
+            Assert.NotEmpty(endpoints);
+
+            var endpoint = endpoints.SingleOrDefault(e => string.Equals(_context.OpcUaEndpointId, e.Id));
```

Comment:
I would use Single instead so that we know exactly why test failed if this id is not present

Reply:
Clarified with assert message.